pyexchange/fix.py

```diff
@@ -208,6 +209,41 @@ def wait_for_get_orders_response(self) -> List[simplefix.FixMessage]:
         messages = self.caller_loop.run_until_complete(self._wait_for_get_orders_response())
         return messages
 
+    async def _listen_for_cancellations(self) -> List[simplefix.FixMessage]:
+        """
+            Iterate through application messages and check to see that we haven't received any order cancellation messages.
+            If a message is recieved, it will be queued and checked periodically as part of the keeper lifecycle.
+        """
+        cancellation_messages = []
+
+        cancel_message_types = [simplefix.EXECTYPE_CANCELED, simplefix.EXECTYPE_EXPIRED]
+
+        while True:
+            if not self.application_messages.empty():
+                message = self.application_messages.get()
+                assert isinstance(message, simplefix.FixMessage)
+
+                # handle message cancellation
+                if message.get(simplefix.TAG_MSGTYPE) == simplefix.MSGTYPE_EXECUTION_REPORT:
+
+                    if message.get(simplefix.TAG_EXECTYPE) in cancel_message_types:
+                        order_to_cancel_id = f"{message.get(simplefix.TAG_ORDERID)}|{message.get(simplefix.TAG_CLORDID)}"
+
+                        if message.get(b'5001') == '4'.encode('utf-8'):
```

Comment:
This tag is likely specific to ErisX.  If we add support for another exchange, we'll need to move this logic into `ErisxFix` class.

Reply:
That makes sense to me, I was also thinking about moving all of these `wait_for_response` methods to the ErisXFix class in erisx.py entirely since the logic is largely exchange specific.